docs/user-guide/src/bmo/firmware_components.md

```diff
@@ -0,0 +1,159 @@
+# Firmware Components
+
+Metal3 supports modifying firmware components of the bare metal hosts.
+This feature can be used to update the firmware for BIOS or BMC.
+The corresponding functionality in Ironic is called
+[Firmware Updates][1].
+
+Updating the firmware components is only supported for Redfish based drivers
+(see [supported hardware](./supported_hardware.md)).
+
+## HostFirmwareComponents Resources
+
+A `HostFirmwareComponents` resource can be created manually or automatically
+for each host that supports firmware components with the same name and in the
+same namespace as host.  BareMetal Operator puts the current components
+information in the `status.components` field:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HostFirmwareComponents
+metadata:
+  creationTimestamp: "2024-08-08T16:44:34Z"
+  generation: 1
+  name: worker-0
+  namespace: my-cluster
+  ownerReferences:
+  - apiVersion: metal3.io/v1alpha1
+    kind: BareMetalHost
+    name: worker-0
+    uid: bef07c46-0674-4c65-8613-d29920e207b1
+  resourceVersion: "21527"
+  uid: 1f9d5b76-5b17-44a1-84f8-7242daafc51d
+spec:
+  updates: []
+status:
+  components:
+  - component: bios
+    currentVersion: 2.3.5
+    initialVersion: 2.3.5
+  - component: bmc
+    currentVersion: 6.10.30.00
+    initialVersion: 6.10.30.00
+  conditions:
+  - lastTransitionTime: "2024-08-08T16:44:35Z"
+    message: ""
+    observedGeneration: 1
+    reason: OK
+    status: "True"
+    type: Valid
+  - lastTransitionTime: "2024-08-08T16:44:35Z"
+    message: ""
+    observedGeneration: 1
+    reason: OK
+    status: "False"
+    type: ChangeDetected
+  lastUpdated: "2024-08-08T16:44:35Z"
+```
+
+This example was taken from a real hardware and was automatically generated:
+
+- The `spec.updates` list is empty - no change is requested by the user.
+
+- The `status.updates` will only be present when `spec.updates` is not empty
+  and an update was executed.
+
+- The `status.components` information is populated with the current values
+  detected by Ironic. If an update is executed the newer information will
+  be available when the host moves from `available`.
+
+- The `Valid` condition is `True`, which means that `spec.updates` are valid,
+  since it was automatically generated we allow the `spec.updates` to be an
+  empty list. The condition will be set to `False` if any value in
+  `spec.updates` fails validation.
+
+- The `ChangeDetected` condition is `False`, which means that the information
+  provided in status matches the information from Ironic and from
+  `spec.updates`. This condition will be set to `True` after you modify
+  `spec.updates` until the change is reflected in `status.updates`.
+
+**Warning:** The components in status are only updated on
+enrollment and provisioning (if an update was executed).
```

Comment:
I think I know what you're trying to say here, but let's move the focus a bit on emphasizing that we don't periodically pull firmware versions from hardware.